readium/navigator/src/main/java/org/readium/r2/navigator/epub/EpubNavigatorViewModel.kt

```diff
@@ -205,7 +206,9 @@ internal class EpubNavigatorViewModel(
         val href = url.toString()
         val link = internalLinkFromUrl(href)
         if (link != null) {
-            _events.send(Event.GoTo(link))
+            if (listener?.onShouldJumpToLink(link) == true) {
+                _events.send(Event.GoTo(link))
```

Comment:
For the sake of consistency, I guess we could have a specialized event and do the check in `EpubNavigatorFragment`.

Reply:
So that the ViewModel doesn't need to "know" about the listener? The Event.GoTo is only used here, so I don't know if we'd need to have a different one? 

I think the idea here (from @mickael-menu) was to ensure that if anyone else sent this event we would avoid calling the `onShouldJumpToLink` callback by making this "private" to the ViewModel. 

My first thought was to call `onShouldJumpToLink` in the EPubNavigatorFragment when it handles events.

Reply:
Ok! I would follow your first thought, renaming `Event.GoTo` to `Event.InternalLinkClicked`, both approaches are fine though.

Reply:
`Event` is poorly named (my fault...), it's more like commands that the view model send to the navigator. Initially `GoTo` was meant to be a general go to command that could be used for other reasons the view model might want to move. But it looks like it's only being used for internal links, so renaming it to `Event.OpenInternalLink` would be fine. (And maybe renaming `Event` to `Command`?).

Reply:
If we call it `OpenInternalLink` - should we still delegate to the ViewModel to call the listener's `onShouldOpenLink`()?

I'll leave `Event` -> `Command` out for now :)

Reply:
All of that is private, so it doesn't matter much after all.